### ISSUE TITLE
examples/{u,glx}play: incorrect stride on some sources

### ISSUE THREAD

**Issue by tmatth** [Upipe/upipe#414]
To reproduce, run `./uplay ed.480p.ts` to get this type of output:
![uplay-ed](https://user-images.githubusercontent.com/66862/35523519-09332f90-04ed-11e8-91ee-2706233378c8.png)

or similarly (but in monochrome) for `./glxplay ed.480p.ts`:
![glxplay-ed](https://user-images.githubusercontent.com/66862/35523563-229e24ee-04ed-11e8-8263-4f1f5f0d79d9.png)

Sample:
http://people.videolan.org/~tmatth/ed.480p.ts

**Comment by funman** [comment]
It works with uplay -s 856x480 although the file is 854x480

**Comment by funman** [comment]
glxplay is different because it uses rgb24 rather than rgb15

**Comment by funman** [comment]
```
diff --git a/lib/upipe-av/upipe_avcodec_decode.c b/lib/upipe-av/upipe_avcodec_decode.c
index bf674ca8..46b4c969 100644
--- a/lib/upipe-av/upipe_avcodec_decode.c
+++ b/lib/upipe-av/upipe_avcodec_decode.c
@@ -295,8 +295,8 @@ static int upipe_avcdec_get_buffer_pic(struct AVCodecContext *context,
     }
 
     UBASE_FATAL(upipe, uref_pic_flow_set_align(flow_def_attr, align))
-    UBASE_FATAL(upipe, uref_pic_flow_set_hsize(flow_def_attr, context->width))
-    UBASE_FATAL(upipe, uref_pic_flow_set_vsize(flow_def_attr, context->height))
+    UBASE_FATAL(upipe, uref_pic_flow_set_hsize(flow_def_attr, context->coded_width))
+    UBASE_FATAL(upipe, uref_pic_flow_set_vsize(flow_def_attr, context->coded_height))
     UBASE_FATAL(upipe, uref_pic_flow_set_hsize_visible(flow_def_attr, context->width))
     UBASE_FATAL(upipe, uref_pic_flow_set_vsize_visible(flow_def_attr, context->height))
     struct urational fps;
```

**Comment by funman** [comment]
@cmassiot ?

**Comment by tmatth** [comment]
That change works as expected here.

**Comment by nto** [comment]
Additionally, shouldn't upipe-gl use hsize_visible and vsize_visible somewhere to crop the visible surface?

**Comment by cmassiot** [comment]
I don't think this is the appropriate patch for several reasons :
- hsize_visible and vsize_visible are currently unused in upipe and I'm thinking of deprecating them. It is due to the fact that it is an internal codec value, and it is equivalent to doing ubuf_pic_alloc(codec_size) then ubuf_pic_resize(visible_size)
- it doesn't fix the real bug in upipe-gl which is that it considers that width == stride.
The attached patch fixes the last issue. It is wrong however as pixels beyond the width shouldn't be displayed, but I don't know how to do it with the OpenGL API.
[upipe.txt](https://github.com/Upipe/upipe/files/1822160/upipe.txt)

**Comment by cmassiot** [comment]
Fixed by ec36886f010f1eea466782ccf522da535aca1243
